### ISSUE TITLE
Rule Change: report WebAssembly in no-obj-calls

### ISSUE THREAD

**Issue by idkned** [eslint/eslint#20955]
### What rule do you want to change?

no-obj-calls

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
// Incorrect (this throws a TypeError at runtime)
WebAssembly();
new WebAssembly();

const x = WebAssembly;
x();

// Correct
WebAssembly.compile(bytes);
new WebAssembly.Module(bytes);
```

### What does the rule currently do for this code?

Currently, the rule ignores WebAssembly() and new WebAssembly(), so no errors or warnings are reported. This allows broken code (which throws a TypeError at runtime) to silently pass linting.

### What will the rule do after it's changed?

After the change, the rule will accurately report WebAssembly() and new WebAssembly() as errors, just like it already does for Math(), JSON(), and Temporal(). Valid usages like WebAssembly.compile() will remain unaffected.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

WebAssembly is a global namespace object with no [[Call]]/[[Construct]] internal methods, so it belongs to the same set the rule already covers (Atomics, JSON, Math, Reflect, Intl, Temporal). This mirrors how Atomics was added (#12234) and Temporal in v10.2.0 — both treated as semver-minor. Implementation is a one-line addition to nonCallableGlobals; the existing ReferenceTracker logic handles call/construct/reference cases automatically.

**Comment by mdjermanovic** [comment]
Thanks for the issue, but `WebAssembly` is not part of the ECMAScript Specification. while the `no-obj-calls` rule only covers ECMAScript standard built-in globals, so we can't accept this proposal.
